### PR TITLE
Add `help`-like functionality upon a parameterless `spine` command usage.

### PR DESCRIPTION
**Pull request by serhii-lekariev**
This PR closes https://github.com/SpineEventEngine/cli/issues/11.

Before, when using `spine` command, a stack-trace was printed, since there was no fitting command.

Now, using `spine` without parameters prints all of the available commands. 

This PR also updates Spine version to `1.0.0-pre6`, thus modifying the way aggregate lookup now works - instead of parsing `KnownTypes`, `FileDescriptors` are now scanned for messages that are aggregates.

### CODE REVIEW

File: command-line/src/main/java/io/spine/tools/cli/CliCommand.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.spine.tools.cli;
+
+/**
+ * A CLI command.
+ *
+ * <p>Behaviour of the command is defined by the {@link this#run()} method inherited from the
```
Comment (dmdashenkov) [inline command-line/src/main/java/io/spine/tools/cli/CliCommand.java]: For such documentation, override method `run` and document it.

Also, does this syntax `{@link this#run()}` actually work?

File: command-line/src/main/java/io/spine/tools/cli/ProjectSetup.java [hunk 2]
```diff
@@ -138,4 +138,23 @@ private void setProjectDestination(File destination) {
                                                    "It can only be set once.");
         }
     }
+
+    @Override
+    public String name() {
+        return "setup";
```
Comment (dmdashenkov) [inline command-line/src/main/java/io/spine/tools/cli/ProjectSetup.java]: Please extract a constant.

File: command-line/src/main/java/io/spine/tools/cli/ProjectSetup.java [hunk 3]
```diff
@@ -138,4 +138,23 @@ private void setProjectDestination(File destination) {
...
+    }
+
+    @Override
+    public String usageInstructions() {
+        final String newLine = System.lineSeparator();
+        String result = "Usage: spine setup -g=<groupId> -n=<projectName> [-p=<projectDestination>]" + newLine +
+        "  -g, --group=<groupId>      The ID of a gradle group of the new project." + newLine +
+        "  -n, --name=<projectName>   A name of the new project." + newLine +
+        "  -p, --path=<projectDestination>" + newLine +
```
Comment (dmdashenkov) [inline command-line/src/main/java/io/spine/tools/cli/ProjectSetup.java]: It is strange that we do this manually. Can these instructions be constructed by the CLI library itself?

**Comment by serhii-lekariev** [comment]
@dmdashenkov, PTAL.

File: code-generation/src/main/java/io/spine/tools/cli/generation/AggregateLookup.java [hunk 4]
```diff
@@ -21,40 +21,46 @@
...
  * An entity that looks for aggregate declarations in
  * {@linkplain com.google.protobuf.DescriptorProtos.FileDescriptorSet descriptor set files}.
  */
 final class AggregateLookup {
 
+    @SuppressWarnings("DuplicateStringLiteralInspection")
+    /* Different semantics - this is a suffix that collected aggregates are going to have. */
     private static final String AGGREGATE_SUFFIX = "Aggregate";
-    private final KnownTypes knownTypes;
+    private final ImmutableSet<FileDescriptorProto> filesToParse;
```
Comment (dmdashenkov) [inline code-generation/src/main/java/io/spine/tools/cli/generation/AggregateLookup.java]: Please think of a better name. An `AggregateLookup` does not **parse**, it **searches**.

File: code-generation/src/main/java/io/spine/tools/cli/generation/AggregateLookup.java [hunk 5]
```diff
@@ -69,11 +75,28 @@ private AggregateLookup(KnownTypes knownTypes) {
      *         the use of this method.
      */
     public static AggregateLookup fromDescriptorSetFile(File descriptorSetFile) {
-        MoreKnownTypes.extendWith(descriptorSetFile);
-        AggregateLookup result = new AggregateLookup(KnownTypes.instance());
+        FileSet fileSet = from(descriptorSetFile);
```
Comment (dmdashenkov) [inline code-generation/src/main/java/io/spine/tools/cli/generation/AggregateLookup.java]: This will not work. Let's discuss.

**Comment by serhii-lekariev** [comment]
@dmdashenkov, PTAL again.

File: code-generation/src/main/java/io/spine/tools/cli/generation/AggregateLookup.java [hunk 6]
```diff
@@ -60,43 +63,64 @@
...
+     * Obtains file descriptors from the specified file.
+     *
+     * <p>Scans the file extracting the names of files that contain actual descriptors, then
+     * obtains a file descriptor for each of the files.
+     *
+     * @param file
+     *         file that contains the names of descriptor set files.
+     * @return a set of file descriptors described by the specified file
+     */
+    private static ImmutableSet<FileDescriptorProto> fileDescriptors(File file) {
```
Comment (dmdashenkov) [inline code-generation/src/main/java/io/spine/tools/cli/generation/AggregateLookup.java]: `loadAllDescriptors(File referenceFile)`

File: code-generation/src/main/java/io/spine/tools/cli/generation/AggregateLookup.java [hunk 7]
```diff
@@ -164,6 +185,7 @@ private static boolean isAbsolute(String name) {
         return name.startsWith(".");
     }
 
+    /** Obtains the type of validating builder of the entity in the specified Java package. */
     private static VBuilderType vBuilder(String javaPackage, String entityName) {
```
Comment (dmdashenkov) [inline code-generation/src/main/java/io/spine/tools/cli/generation/AggregateLookup.java]: It is not an entity, in Spine terms.

File: code-generation/src/main/java/io/spine/tools/cli/generation/AggregateLookup.java [hunk 8]
```diff
@@ -21,59 +21,106 @@
...
 
 /**
  * An entity that looks for aggregate declarations in
  * {@linkplain com.google.protobuf.DescriptorProtos.FileDescriptorSet descriptor set files}.
  */
 final class AggregateLookup {
 
+    @SuppressWarnings("DuplicateStringLiteralInspection")
+    /* Different semantics - this is a suffix that collected aggregates are going to have. */
     private static final String AGGREGATE_SUFFIX = "Aggregate";
```
Comment (dmdashenkov) [inline code-generation/src/main/java/io/spine/tools/cli/generation/AggregateLookup.java]: Appending a postfix to a message type name should not be a part of `AggregateLookup`. My guess is that it belongs to `AggregateDefinition`.

**Pull request merged**